src/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/repair/sequencer/SequencerRepair.java

```diff
@@ -179,23 +215,33 @@ protected StepStatus businessExecute() {
         List<RepairPatch> listPatches = new ArrayList<>();
         JsonArray toolDiagnostic = new JsonArray();
 
-        boolean success = false;
-        for (SequencerResult result : sequencerResults) {
-            JsonObject diag = new JsonObject();
-
-            diag.addProperty("success", result.isSuccess());
-            diag.addProperty("message", result.getMessage());
-            diag.addProperty("warning", result.getWarning());
-            toolDiagnostic.add(diag);
-
-            if (result.isSuccess()) {
-                success = true;
-                List<String> diffs = result.getDiffs();
-                for (String diff : diffs) {
-                    RepairPatch patch = new RepairPatch(this.getRepairToolName(), result.getBuggyFilePath(), diff);
-                    listPatches.add(patch);
-                }
-            }
+        listPatches = sequencerResults.stream().flatMap( result -> {
+            JsonObject diagnostic = new JsonObject();
+
+            diagnostic.addProperty("success", result.isSuccess());
+            diagnostic.addProperty("message", result.getMessage());
+            diagnostic.addProperty("warning", result.getWarning());
+            toolDiagnostic.add(diagnostic);
+
+            List<String> diffs = result.getDiffs();
+
+            Stream<RepairPatch> patches = diffs.stream()
+                .map(diff -> new RepairPatch(this.getRepairToolName(), result.getBuggyFilePath(), diff))
+//                .filter(patch -> {
+//                    Properties properties = new Properties();
+//                    properties.setProperty(MavenHelper.SKIP_TEST_PROPERTY, "true");
+//                    return testMavenGoal(patch, "package", properties); })
+                .filter(patch -> {
+                    Properties properties = new Properties();
+                    return testMavenGoal(patch, "test", properties);
+                });
```

Comment:
I will leave for now a local "hard" `mvn test` for each patch. In a later PR I will implement the "soft" condition (the line is no longer reported as suspicious), since right now I don't have a reliable line detection tool.

Reply:
ok for me